### PR TITLE
Refactors antag datum hud adding and removing procs

### PR DESCRIPTION
**Pull request by SteelSlayer**
## About The Pull Request
This removes all of the hud adding and removing procs that were used to in antag datums. Those being `update_traitor_icons_added`, `update_traitor_icons_removed` and so forth.

Now we have two new procs in the base `antag_datum.dm` file called `add_antag_hud` and `remove_antag_hud` which takes arguments from child proc calls and dynamically assigns the hud to the owner of the datum, based on what you pass into it.

This should cut down on a lot of needless copy and paste of creating the same hud proc, with a different name in each of the antag datum files.

Tested out all the antag's huds and also tested on whether cloning would transfer the hud properly. Seems to work fine, though I might have missed something.

EDIT: Went ahead and put a single proc that handles adding/removing antag clown's clumsy mutation into antag_datum.dm. Cuts down on more copy and paste.

## Changelog
:cl:
code: Cleans up some scattered hud removal procs and puts them into their respective antag datum file so when the player loses the antag status, they lose the hud
refactor: Refactors antag hud adding and removing procs. Moves them into the base antag datum file. The new procs can dynamically assign a hud from given arguments. Cuts down on copy and paste
refactor: Refactors the clown mutation adding and removing procs for antags. Merges it into a single proc in the base antag datum file.
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
wonderful, love it!

**Comment by SteelSlayer** [comment]
Trying to unbork travis

File: code/modules/antagonists/wizard/wizard.dm [hunk 1]
```diff
@@ -152,12 +153,12 @@
 
 /datum/antagonist/wizard/apply_innate_effects(mob/living/mob_override)
 	var/mob/living/M = mob_override || owner.current
-	update_wiz_icons_added(M, wiz_team ? TRUE : FALSE) //Don't bother showing the icon if you're solo wizard
```
Comment (AnturK) [inline code/modules/antagonists/wizard/wizard.dm]: Not keeping this ?

File: code/modules/antagonists/_common/antag_datum.dm [hunk 2]
```diff
@@ -59,6 +61,31 @@ GLOBAL_LIST_EMPTY(antagonists)
...
+	var/datum/atom_hud/antag/hud = GLOB.huds[antag_hud_type]
+	hud.join_hud(mob_override)
+	set_antag_hud(mob_override, null)
+
+// Handles adding and removing the clumsy mutation from clown antags. Gets called in apply/remove_innate_effects
+/datum/antagonist/proc/handle_clown_mutation(mob/living/mob_override, message)
+	var/mob/living/carbon/human/H = mob_override
+	if(H.mind.assigned_role == "Clown")
+		if(H && istype(H))
+			if(H.dna.check_mutation(CLOWNMUT)) // They're a clown becoming an antag, remove clumsy
```
Comment (AnturK) [inline code/modules/antagonists/_common/antag_datum.dm]: I'd prefer if it was explicit if we're adding or removing it, nothing stops clown that lost it before in some way getting antag later.

**Review by AnturK — CHANGES_REQUESTED** [review]
Looks alright, just some nitpicks.

File: code/modules/antagonists/wizard/wizard.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/modules/antagonists/wizard/wizard.dm]: As far as I can tell, the extra check in the original proc doesn't do anything. Original proc:
```
/datum/antagonist/wizard/proc/update_wiz_icons_added(mob/living/wiz,join = TRUE)
	var/datum/atom_hud/antag/wizhud = GLOB.huds[ANTAG_HUD_WIZ]
	wizhud.join_hud(wiz)
	set_antag_hud(wiz, hud_version)
```
So we have the `join` variable but there's never any checks made with it. I'm still getting the wizhud as a solo wizard on the master branch.

File: code/modules/antagonists/_common/antag_datum.dm (hunk 2 above)
Comment (SteelSlayer) [reply code/modules/antagonists/_common/antag_datum.dm]: Should I make two seperate procs to add and remove?

File: code/modules/antagonists/wizard/wizard.dm (hunk 1 above)
Comment (AnturK) [reply code/modules/antagonists/wizard/wizard.dm]: All good then.

File: code/modules/antagonists/_common/antag_datum.dm (hunk 2 above)
Comment (AnturK) [reply code/modules/antagonists/_common/antag_datum.dm]: That or pass in the new value, either way works.

Comment (SteelSlayer) [reply code/modules/antagonists/_common/antag_datum.dm]: Which new value do you mean?

Comment (AnturK) [reply code/modules/antagonists/_common/antag_datum.dm]:
```
handle_clown_mutation(x,y,remove=TRUE)
  if(add)
    add mutation
  else
    remove mutation
``` Or whatever configuration of parameters will be used most often.

Comment (SteelSlayer) [reply code/modules/antagonists/_common/antag_datum.dm]: Alright, made some changes, hopefully it makes sense

File: code/modules/antagonists/_common/antag_datum.dm [hunk 3]
```diff
@@ -75,16 +75,15 @@ GLOBAL_LIST_EMPTY(antagonists)
...
-	if(H.mind.assigned_role == "Clown")
-		if(H && istype(H))
-			if(H.dna.check_mutation(CLOWNMUT)) // They're a clown becoming an antag, remove clumsy
-				if(!silent && message)
-					to_chat(H, "<span class='boldnotice'>[message]</span>")
-				H.dna.remove_mutation(CLOWNMUT)
-			else
-				H.dna.add_mutation(CLOWNMUT) // We're removing their antag status, add back clumsy
+	if(H && H.mind.assigned_role == "Clown")
+		if(removing) // They're a clown becoming an antag, remove clumsy
```
Comment (AnturK) [inline code/modules/antagonists/_common/antag_datum.dm]: You should still check if they are actually human

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/_common/antag_datum.dm (hunk 3 above)
Comment (SteelSlayer) [reply code/modules/antagonists/_common/antag_datum.dm]: oops, you're right, putting that check back

**Pull request merged**